lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -1186,4 +1188,23 @@ public function quoteStringLiteral($str)
 
         return parent::quoteStringLiteral($str);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultValueDeclarationSQL($field)
+    {
+        if (self::isSerialField($field) && empty($field['notnull'])) {
+            return '';
+        }
+
+        return parent::getDefaultValueDeclarationSQL($field);
+    }
+
+    private static function isSerialField($field)
```

Comment:
Missing type declarations (parameters and return)

Reply:
Also, any reason to make this static?

Reply:
> Missing type declarations (parameters and return)

Left out the parameter type declaration because the method from which this is called doesn't have one either (`getDefaultValueDeclarationSQL`). `$field` theoretically should be `array`? Happy to add the type declaration for `$field`, but that was the reason I didn't.

Definitely should have the return type!

> Also, any reason to make this static?

Habit of my, sorry. I tend to make "pure" utility functions static -- since nothing from the platform's instance state is required, I made it static. Want me to make it non-static?

Reply:
I got your point. I don't see much difference in this case, since the caller is already in the scope of the instance. I'd keep it non-static here but was just curious 😄